src/compiler/program.ts

```diff
@@ -1005,15 +1019,30 @@ namespace ts {
         const filesByNameIgnoreCase = host.useCaseSensitiveFileNames() ? createFileMap<SourceFile>(fileName => fileName.toLowerCase()) : undefined;
 
         if (!tryReuseStructureFromOldProgram()) {
+            forEach(rootNames, name => processRootFile(name, /*isDefaultLib*/ false));
+
             // load type declarations specified via 'types' argument
-            if (options.types && options.types.length) {
-                const resolutions = resolveTypeReferenceDirectiveNamesWorker(options.types, /*containingFile*/ undefined);
-                for (let i = 0; i < options.types.length; i++) {
-                    processTypeReferenceDirective(options.types[i], resolutions[i]);
+            let typeReferences: string[];
+            if (options.types) {

```

Comment:
it won't work in VS since these files won't be discovered during preprocessing (both `then` and `else` statements are affected)